src/DocumentFeedMigrator.cs

```diff
@@ -67,10 +71,45 @@ public DocumentFeedMigrator(Container containerToStoreDocuments)
                     log.LogMetric($"Failed items with StatusCode {failedMetric.Key}", failedMetric.Value);
                     totalFailedItems += failedMetric.Value;
                 }
-
                 log.LogMetric("Documents migrated", documents.Count - totalFailedItems);
                 log.LogMetric("Migration Time in ms", stopwatch.ElapsedMilliseconds);
             }
+
+        }
+
+        public Document MapPartitionKey(Document doc)
+        {
+            if(isSyntheticKey)
+            {
+                doc = CreateSyntheticKey(doc);
+            }
+            else
+            {
+                doc.SetPropertyValue(targetPartitionKeyAttribute, doc.GetPropertyValue<string>(sourcePartitionKeyMapping));
+            }
+            return doc;
+        }
+
+        public Document CreateSyntheticKey(Document doc)
+        {
+            StringBuilder syntheticKey = new StringBuilder();
+            string[] sourceAttributeArray = sourcePartitionKeyMapping.Split(',');
```

Comment:
I wonder, will this work with nested PKs? like `property.nestedProperty`?

Reply:
I am not sure, but in practice I think nested pk is rare.